src/jit/codegenarm64.cpp

```diff
@@ -2820,6 +2830,11 @@ void CodeGen::genCodeForCpObj(GenTreeObj* cpObjNode)
         assert(tmpReg2 != REG_WRITE_BARRIER_SRC_BYREF);
     }
 
+    if (cpObjNode->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
A comment stating that we place a full memory barrier before and after the cpobj operation, should be added here